lib/usb_connection.js

```diff
@@ -189,11 +190,10 @@ USB.Connection.prototype.end = function() {
     // Tell the USB daemon to end all processes active
     // on account of this connection
     Daemon.deregister(self, function(err) {
-      self._close();
       if (err) {
         reject(err);
       } else {
-        resolve();
+        self._close(resolve());
```

Comment:
Thanks, I think I fixed it with https://github.com/tessel/t2-cli/commit/885f385804e69533b6c86fa623adb412acbeb474. It was actually concealing a bigger problem with the tests!